packages/shared/lib/contexts/onboarding/stores/onboarding-secret-manager.store.ts

```diff
@@ -1,25 +1,41 @@
-import { onboardingProfile } from '@contexts/onboarding/stores'
+import { onboardingProfile, updateOnboardingProfile } from '@contexts/onboarding/stores'
 import { api } from '@core/api'
-import { SecretManager, SecretManagerType } from '@iota/sdk'
-import { USE_LEDGER_SIMULATOR } from 'shared/lib/core/ledger'
-import { getStorageDirectoryOfProfile, ProfileType } from 'shared/lib/core/profile'
+import { SecretManager } from '@iota/sdk'
+import { verifyMnemonic } from 'shared/lib/core/secret-manager'
 import { get, writable, Writable } from 'svelte/store'
 
 export const onboardingProfileSecretManager: Writable<SecretManager | null> = writable(null)
 
 export async function buildOnboardingSecretManager(): Promise<void> {
     const profile = get(onboardingProfile)
     if (profile) {
-        const { id, type, strongholdPassword } = profile
+        const { strongholdPassword, secretManagerOptions, mnemonic } = profile
+        const mnemonicStringified = mnemonic?.join(' ') ?? ''
 
-        const storagePath = await getStorageDirectoryOfProfile(id)
-        const secretManagerOptions = getSecretManagerFromProfileType(type, {
-            storagePath,
-            strongholdPassword,
-        })
+        if (!strongholdPassword || !secretManagerOptions || !mnemonic) {
+            return
+        }
 
+        // 1. Create SecretManager
         const secretManager = await api.createSecretManager(secretManagerOptions)
 
+        // 2. Load the stronghold password specified in the onboarding if necessary
+        if (strongholdPassword) {
+            await secretManager.setStrongholdPassword(strongholdPassword)
+        }
+
+        // 3. Verify Mnemonic
+        await verifyMnemonic(mnemonicStringified)
+
+        // 4. Store Mnemonic
+        await secretManager.storeMnemonic(mnemonicStringified)
+
+        // 5. Generate address
+        const address = (await secretManager.generateEd25519Addresses({}))[0]
+
+        updateOnboardingProfile({
+            address,
+        })
```

Comment:
the address does not seem to be necessary, the wallet can be initialized with the secret manager and the correct bip path, and then the SDK should be able to generate the address by itself

Reply:
Just removed the address creation and instead I will now pass the password to the wallet options the first time the wallet is created